### PR TITLE
Fix typos discovered by codespell

### PR DESCRIPTION
**Pull request by cclauss**
% `pipx run codespell --ignore-words-list=cheking,clen,fo,ned,reurn,smoe,tihs`
% `pipx run codespell --ignore-words-list=cheking,clen,fo,ned,reurn,smoe,tihs --write-changes`
https://pypi.org/project/codespell

### CODE REVIEW

**Comment by facelessuser** [comment]
Yep, these are some good catches. I'm scanning strings right now, and things like `wit` and `with` are examples of typos that appear as correctly spelled words, but with context are clearly typos. Thanks for the pull!

File: tests/filters/test_cpp.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def test_cpp(self):
                 uint8_t tsdd = 3;
                 // {}
                 // {} \
-                reurn tsdd;
```
Comment (facelessuser) [inline tests/filters/test_cpp.py:48]: Please undo the `reurn` changes, we are intentionally testing for this mispelling.

**Review by facelessuser — COMMENTED** [review]
In order for the tests to pass, we need test content with misspelling to be unaltered.

File: tests/filters/test_cpp.py (hunk 1 above)
Comment (cclauss) [reply tests/filters/test_cpp.py:48]: OK!

**Comment by facelessuser** [comment]
Okay, things are looking better now. I made a mistake earlier, I am _not_ scanning strings, but I should consider it in the future as that would have caught the string misspellings. Anyways, I think this looks good now.

@gir-bot lgtm

**Pull request merged**